### PR TITLE
remove some uncaught promise warnings during peerconnection setup

### PR DESCRIPTION
**Pull request by trevj**
This should eliminate the annoying warnings we're seeing when peerconnection setup fails in the UI:
https://github.com/uProxy/uproxy/issues/997

[Review on Reviewable](https://reviewable.io/reviews/uproxy/uproxy-lib/135)

### CODE REVIEW

File: src/webrtc/peerconnection.ts [hunk 1]
```diff
@@ -209,6 +209,8 @@ export class PeerConnectionClass implements PeerConnection<SignallingMessage> {
     // Once connected, add to global listing. Helpful for debugging.
     this.onceConnected.then(() => {
       PeerConnectionClass.peerConnections[this.peerName_] = this;
+    }, (e:Error) => {
+      // No-op: this is just to avoid uncaught promise warnings.
```
Comment (bemasc) [inline src/webrtc/peerconnection.ts]: We have a logger.  Might as well use it.

Comment (trevj) [inline src/webrtc/peerconnection.ts]: Done!

**Comment by trevj** [comment]
PTAL.

**Comment by iislucas** [comment]
Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy-lib/135)

Reviewed files:
- package.json @ r2
- src/samples/copypaste-freedom-chat/freedom-module.ts @ r1
- src/samples/simple-freedom-chat/freedom-module.ts @ r1
- src/webrtc/peerconnection.ts @ r2

---

LGTM! Thanks for these cleanups!

Add a note on how it was tested (and I think it may need a merge from head of dev)

---

**Comment by trevj** [comment]
So, `grunt test` passes and I verified that the uncaught exception warning disappears by hacking `onIceConnectionStateChange_` (in `peerconnection.ts`) to (unconditionally) call `this.closeWithError_`. This has the effect of making `onceConnected` reject.

**Pull request merged**